lib/restaurant.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'message_sender'
+class Restaurant
+
+  attr_accessor :menu, :order, :total_amount, :text_is_sent
+
+  def initialize
+    @menu = {
+      :Beef_BBQ => 15,
+      :Chicken_Supreme => 16,
+      :Hawaiian => 14,
+      :Margherita => 12,
+      :Pepperoni => 16,
+      :Texan_BBQ => 15,
+      :Tuna_Sweetcorn => 17,
+      :Vegetable => 16
+      }
+      @order = []
+      @total_amount = 0
+      @text_is_sent = Message_sender.new
+    end
+
+    def dishes_can_be_selected dish, quantity
+    quantity.times do
+      order << menu.select {|x,y| x == dish}
+    end
+  end
+
+    def give_total_amount
+    total_amount = order.map{|item| item.values}.flatten.inject{|sum,x| sum + x }
```

Comment:
This line is too hard to read.  Avoid 'clever' code - it's a prime place for subtle bugs to hide.  Each of the steps in this line of code should be refactored into their own methods so the method names can encapsulate the logic.  That way it reads like a sentence, rather than this, which is impossible to read without running the logic in your head.